### PR TITLE
Support "At least 1 uppercase or lowercase alphabet" in password policy

### PR DESCRIPTION
**Pull request by tung2744**
ref https://github.com/authgear/authgear-server/issues/3079

### CODE REVIEW

**Comment by louischan-oursky** [comment]
Please ignore the complexity lint error.

File: pkg/lib/authn/authenticator/password/checker.go [hunk 1]
```diff
@@ -297,6 +312,9 @@ func (pc *Checker) PasswordRules() string {
 	if pc.PwLowercaseRequired {
 		builder.WriteString("required: lower; ")
 	}
+	if pc.PwAlphabetRequired {
+		builder.WriteString("required: alphabet; ")
```
Comment (louischan-oursky) [inline pkg/lib/authn/authenticator/password/checker.go:316]: alphabet is not a builtin character class. https://developer.apple.com/documentation/security/password_autofill/customizing_password_autofill_rules

I think we need to use **Custom Character Class** here.

File: pkg/lib/authn/authenticator/password/checker.go [hunk 2]
```diff
@@ -59,6 +59,15 @@ func checkPasswordLowercase(password string) bool {
 	return false
 }
 
+func checkPasswordAlphabet(password string) bool {
```
Comment (louischan-oursky) [inline pkg/lib/authn/authenticator/password/checker.go:62]: Update checker_test.go to test new changes.

File: portal/src/locale-data/en.json [hunk 3]
```diff
@@ -518,10 +518,11 @@
...
   "PasswordPolicyConfigurationScreen.force-change.label": "Force password change on next login if it does not meet the policy requirements",
   "PasswordPolicyConfigurationScreen.min-length.label": "Minimum password length",
-  "PasswordPolicyConfigurationScreen.require-digit.label": "Requires at least 1 digit",
-  "PasswordPolicyConfigurationScreen.require-lowercase.label": "Requires at least 1 lowercase character",
-  "PasswordPolicyConfigurationScreen.require-uppercase.label": "Requires at least 1 uppercase character",
-  "PasswordPolicyConfigurationScreen.require-symbol.label": "Requires at least 1 symbol",
+  "PasswordPolicyConfigurationScreen.require-digit.label": "Requires at least 1 digit (0-9)",
+  "PasswordPolicyConfigurationScreen.require-lowercase.label": "Requires at least 1 lowercase character (a-z)",
+  "PasswordPolicyConfigurationScreen.require-uppercase.label": "Requires at least 1 uppercase character (A-Z)",
+  "PasswordPolicyConfigurationScreen.require-alphabet.label": "Requires at least 1 alphabetic character (a-z,A-Z)",
```
Comment (louischan-oursky) [inline portal/src/locale-data/en.json:524]: s/alphabetic/alphabet

File: portal/src/locale-data/en.json [hunk 4]
```diff
@@ -746,6 +746,7 @@
   "PasswordField.min-length": "Minimum {minLength} characters",
   "PasswordField.lowercase-required": "At least 1 lowercase character",
   "PasswordField.uppercase-required": "At least 1 uppercase character",
+  "PasswordField.alphabet-required": "At least 1 alphabetic character",
```
Comment (louischan-oursky) [inline portal/src/locale-data/en.json:749]: s/alphabetic/alphabet

File: resources/authgear/templates/en/translation.json [hunk 5]
```diff
@@ -243,6 +243,7 @@
   "password-policy-minimum-length": "Minimum {length, plural, one{# character} other{# characters}}",
   "password-policy-uppercase": "At least 1 uppercase character",
   "password-policy-lowercase": "At least 1 lowercase character",
+  "password-policy-alphabet": "At least 1 alphabetic character",
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/translation.json:246]: s/alphabetic/alphabet

File: pkg/lib/authn/authenticator/password/checker.go (hunk 1 above)
Comment (tung2744) [reply pkg/lib/authn/authenticator/password/checker.go:316]: @louischan-oursky What about using `special`?

Comment (louischan-oursky) [reply pkg/lib/authn/authenticator/password/checker.go:316]: `special` is mapped to our `symbol` policy.

File: portal/src/locale-data/en.json (hunk 3 above)
Comment (tung2744) [reply portal/src/locale-data/en.json:524]: @louischan-oursky It seems it should be "alphabetic character" instead of "alphabet character"

**Comment by tung2744** [comment]
@louischan-oursky Updated, thanks!

**Pull request merged**